### PR TITLE
Add url locale on stepper

### LINKED ISSUE #67366: Task: Implement locale parameter, so marketing pages can redirect to the right language.
### Details

The entire Stepper should support a `locale` parameter that determines the UI language. 

Especially that Since Stepper caters to unauthenticated users now.

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by renancarvalho**
## Proposed Changes

**For not logged in users**, if there is a `&locale=XX` param in the url, the flow should be loaded in that language.
**For logged in users**, we their own locale.

## Testing Instructions

 - Pull this branch and run yarn start from the root folder
 - Navigate to `/setup/intro?flow=newsletter` or `/setup/intro?flow=link-in-bio`
 - You should see the app in your [default language ](https://wordpress.com/me/account) "Interface Language"
 - Add `&locale=XX` you should see on the specific language if supported  

image

image

image

Related to #67366
Fixes #67366

### CODE REVIEW

File: client/landing/stepper/declarative-flow/newsletter.ts [hunk 1]
```diff
@@ -34,17 +35,18 @@ export const newsletter: Flow = {
 		const { setStepProgress } = useDispatch( ONBOARD_STORE );
 		const flowProgress = useFlowProgress( { stepName: _currentStep, flowName: this.name } );
 		setStepProgress( flowProgress );
+		const urlLocale = useLocale();
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/newsletter.ts]: This doesn't come from the URL, it should be `locale`.

File: client/landing/stepper/declarative-flow/newsletter.ts [hunk 2]
```diff
@@ -34,17 +35,18 @@ export const newsletter: Flow = {
...
 
 		function submit( providedDependencies: ProvidedDependencies = {} ) {
 			recordSubmitStep( providedDependencies, '', _currentStep );
-
+			const localeRoute = urlLocale ? `/${ urlLocale }/` : '';
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/newsletter.ts]: We don't need to pass `en` when it's English.

```suggestion
			const localeRoute = urlLocale && urlLocale !== 'en' ? `/${ urlLocale }/` : '';
```

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 3]
```diff
@@ -32,15 +33,17 @@ export const linkInBio: Flow = {
 		setStepProgress( flowProgress );
 		const siteSlug = useSiteSlug();
 		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
+		const urlLocale = useLocale();
 
 		function submit( providedDependencies: ProvidedDependencies = {} ) {
+			const localeRoute = urlLocale ? `/${ urlLocale }/` : '';
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts]: We don't need to pass `en` when it's English.
```suggestion
			const localeRoute = urlLocale && urlLocale !== 'en' ? `/${ urlLocale }/`
```

**Review by alshakero — CHANGES_REQUESTED** [review]
Slick! Left a few cosmetic comments.

File: client/landing/stepper/declarative-flow/newsletter.ts (hunk 2 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/newsletter.ts]: Indeed, I have this covered on the next commit. Thank you.
🥇

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 4]
```diff
@@ -33,18 +33,23 @@ export const linkInBio: Flow = {
...
 		const siteSlug = useSiteSlug();
 		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
-		const urlLocale = useLocale();
+		const locale = useLocale();
+
+		const getStartUrl = () => {
+			return locale && locale !== 'en'
+				? `/start/account/user/${ locale }?variationName=${ name }&pageTitle=Link%20in%20Bio&redirect_to=/setup/patterns?flow=${ name }`
+				: `/start/account/user?variationName=${ name }&pageTitle=Link%20in%20Bio&redirect_to=/setup/patterns?flow=${ name }`;
+		};
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts:42]: Super nit: this is a perfect place for a full `if...else`. Much nicer than ternaries.

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 5]
```diff
@@ -32,16 +33,23 @@ export const linkInBio: Flow = {
 		setStepProgress( flowProgress );
 		const siteSlug = useSiteSlug();
 		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
+		const locale = useLocale();
+
+		const getStartUrl = () => {
+			return locale && locale !== 'en'
+				? `/start/account/user/${ locale }?variationName=${ name }&pageTitle=Link%20in%20Bio&redirect_to=/setup/patterns?flow=${ name }`
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts:40]: We need to add the `locale` in the redirect_to param as well.

File: client/landing/stepper/declarative-flow/newsletter.ts [hunk 6]
```diff
@@ -34,18 +35,24 @@ export const newsletter: Flow = {
 		const { setStepProgress } = useDispatch( ONBOARD_STORE );
 		const flowProgress = useFlowProgress( { stepName: _currentStep, flowName: this.name } );
 		setStepProgress( flowProgress );
+		const locale = useLocale();
+
+		const getStartUrl = () => {
+			return locale && locale !== 'en'
+				? `/start/account/user/${ locale }?variationName=${ name }&pageTitle=Link%20in%20Bio&redirect_to=/setup/patterns?flow=${ name }`
+				: `/start/account/user?variationName=${ name }&pageTitle=Link%20in%20Bio&redirect_to=/setup/patterns?flow=${ name }`;
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/newsletter.ts:43]: +1

**Review by alshakero — CHANGES_REQUESTED** [review]
Almost there

File: client/landing/stepper/declarative-flow/link-in-bio.ts (hunk 5 above)
Comment (alshakero) [reply client/landing/stepper/declarative-flow/link-in-bio.ts:40]: Might be worth it to create a `URLSearchParams` object.

**Review by alshakero — APPROVED** [review]
Works perfect.

**Pull request merged**